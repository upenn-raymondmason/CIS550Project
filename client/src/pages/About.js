import logo from '../resources/logo.svg';
import '../styles/OuterPages.css';
import TopNavBar from '../components/TopNavBar';
import React from 'react'


export default class About extends React.Component{

  constructor(props) {
    super(props);
    this.state = {
    }
  }
  render() {
    return (
      <div data-testid="AboutBox" id="AboutBox" className="App">
        <TopNavBar></TopNavBar>
        <br></br>
        <h1>About</h1>
        <img src={logo} className="App-logo" alt="logo" />
      </div>
    );
  }
}

