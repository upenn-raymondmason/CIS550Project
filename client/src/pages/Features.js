import logo from '../resources/logo.svg';
import '../styles/OuterPages.css';
import TopNavBar from '../components/TopNavBar';
import React from 'react'


export default class Features extends React.Component{

  constructor(props) {
    super(props);
    this.state = {
    }
  }
  render() {
    return (
      <div className="App">
        <TopNavBar></TopNavBar>
        <br></br>
        <h1>Features</h1>
        <img src={logo} className="App-logo" alt="logo" />
      </div>
    );
  }
}