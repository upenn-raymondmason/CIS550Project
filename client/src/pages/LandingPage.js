import logo from '../resources/logo.svg';
import '../styles/OuterPages.css';
import TopNavBar from '../components/TopNavBar';
import LoginForm from '../components/LoginForm';
import React from 'react'


export default class LandingPage extends React.Component{

  constructor(props) {
    super(props);
    this.state = {
    }
    //this.setUser = this.setUser.bind(this);
  }

  render() {
    return (
      <div data-testid = "landing" className="App">
        <TopNavBar></TopNavBar>
        <div style={{float: 'right', alignSelf: 'right', marginRight: '5vw'}}>
          <LoginForm></LoginForm>
        </div>
        <br></br>
        <img src={logo} className="App-logo" alt="logo" />
      </div>
    );
  }
  
}

