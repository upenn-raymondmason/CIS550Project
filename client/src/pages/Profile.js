
import '../styles/OuterPages.css';
import React from 'react'
import InnerTopNavBar from '../components/InnerTopNavBar'
import {changePassword, deactivateAcc, getUser} from '../fetcher'

export default class Profile extends React.Component{

  constructor(props) {
    super(props);
    this.state = {
      username: '',
      newPassword: '',
      password: '',
      date: '',
    }
    this.handleNewPasswordTyped = this.handleNewPasswordTyped.bind(this);
    this.handlePasswordChange = this.handlePasswordChange.bind(this);
    this.validatePassword = this.validatePassword.bind(this);
  }

  componentDidMount() {
    getUser(sessionStorage.getItem('username')).then(res => {
      this.setState({password : res.result.password});
      this.setState({date : res.result.date});
    });
    this.setState({username : sessionStorage.getItem('username')});
  }

  handleNewPasswordTyped(event) {
    this.setState({ newPassword: event.target.value });
  }

  async handlePasswordChange(event) {
    event.preventDefault();
      if (this.validatePassword()) {
        changePassword(this.state.username, this.state.newPassword)
        .then(res => {
            if (res.error) {
                alert(res.error);
            } else if (res.message === 'success') {
                alert("Password changed!")
                this.setState({password : this.state.newPassword});
            }
        });
  }
  }

  async handleDeactivateAcc(event) {
    event.preventDefault();
    deactivateAcc(this.state.username)
    .then(res => {
      if (res.error) {
        alert(res.error);
      } else if (res.message === 'success') {
        alert("Account Deactivated! Now being Logged out");
        sessionStorage.setItem('username', null);
        window.location.assign('/');
        //Potentially more cleanup required, cannot test safely until backend for cognito handling implemented
      }
    });
  }

  validatePassword() {
    if (this.state.newPassword.length < 8) {
        alert("Password must be at least 8 characters long!");
        return false;
    }

    if (!/\d/.test(this.state.newPassword)) {
        alert("Password must contain digits!");
        return false;
    }

    if (!/^(.*[A-Z].*)$/.test(this.state.newPassword)) {
        alert("Password must contain uppercase characters!");
        return false;
    }

    if (!/^(.*[a-z].*)$/.test(this.state.newPassword)) {
        alert("Password must contain lowercase characters!");
        return false;
    }

    if (!/[ `!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?~]/.test(this.state.newPassword)) {
        alert("Password must contain special characters!");
        return false;
    }

    return true;
  }

//change className later
  render() {
    return (
      <div>
          <InnerTopNavBar></InnerTopNavBar>
        <h1>Profile</h1>
        <p>Username: {this.state.username}</p>
        <p>Password(temporary): {this.state.password}</p>
        <p>Registration Date: {this.state.date}</p>
        <div className='LoginForm'>
                    <h2 className='LoginForm-title'>Change Password</h2>
                    <br></br>
                    <form onSubmit={this.handlePasswordChange}>

                        <input className="LoginForm-input" type="text" name="name" placeholder="..." value={this.state.newPassword} onChange={this.handleNewPasswordTyped} />
                        <br></br>
                        <input className="LoginForm-button" type="submit" value="Add" />
                    </form>
                </div>
                <div className='LoginForm'>
                    <h2 className='LoginForm-title'>Deactivate Account</h2>
                    <br></br>
                    <form onSubmit={this.handleDeactivateAcc}>
                        <input className="LoginForm-button" type="submit" value="Deactivate Account!" />
                    </form>
                </div>
      </div>
    );
  }
}


/*import '../styles/OuterPages.css';
import React from 'react'
import InnerTopNavBar from '../components/InnerTopNavBar'
import {getUser} from '../fetcher'

export default class Profile extends React.Component{

  constructor(props) {
    super(props);
    this.state = {
      date: '',
      password: '',
    }

  }

  componentDidMount() {
    getUser(sessionStorage.getItem('username')).then(res => {
      this.setState({date : res.result.date})
    });
  }

  render() {
    return (
      <div>
          <InnerTopNavBar></InnerTopNavBar>
        <h1>Profile</h1>
        <p>Username: {sessionStorage.getItem('username')}</p>
        <p>Registration Date: {this.state.date}</p>

      </div>
    );
  }
} */

