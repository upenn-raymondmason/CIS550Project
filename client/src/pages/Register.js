import logo from '../resources/logo.svg';
import '../styles/OuterPages.css';
import TopNavBar from '../components/TopNavBar';
import React from 'react'
import SignupForm from '../components/SignupForm';

export default class Register extends React.Component{

  constructor(props) {
    super(props);
    this.state = {
    }
  }
  render() {
    return (
      <div className="App">
        <TopNavBar></TopNavBar>
        <div style={{float: 'right', alignSelf: 'right', marginRight: '5vw'}}>
          <SignupForm></SignupForm>
        </div>
        <br></br>
        <img src={logo} className="App-logo" alt="logo" />
      </div>
    );
  }
}