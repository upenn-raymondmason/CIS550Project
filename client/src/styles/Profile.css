.main {
    background: blue;
    display: inline-block;
    justify-content: center;
    align-items: center;
    width: 90vw;
    height: 80vh;
    margin-top: 1.5rem;
}

.profileContainer {
    background-color: green;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 80%;
}

.profileTitle {
    width: 100%;
    height: 20%;
    background-color: greenyellow;
}

.titleContainer {
    background: linear-gradient(90deg,#34abeb 0%, #00dbde 100%);
    border-radius: 20px;

    margin: 0.5rem auto;
    padding: 0;
    width: 80%;
    height: 80%;
    box-shadow:3px 6px #00aade;
    display: inline-block;
    align-items: center;
    justify-content: start;
}

.titleContainer>p {
    font-size: 28px;
    color: white;
    margin: 0 auto;
}

.fav-title {
    font-weight: bold;
    font-size: 26px;
    color: white;
    height: 5%;
    width: 100%;
    margin: 0.5rem auto;
}

.fav-box {
    background: linear-gradient(90deg,#34abeb 0%, #00dbde 100%);
    border-radius: 20px;

    margin: 0.5rem auto;
    padding: 0;
    width: 45%;
    height: 90%;
    box-shadow:3px 6px #00aade;
    display: inline-block;
    align-items: center;
    justify-content: center;
}

.profilePlayerPane {
    background-color: grey;
    height: 85%;
    width: 100%;
}