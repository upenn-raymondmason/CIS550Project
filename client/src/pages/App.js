import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import React from 'react'
import LandingPage from './LandingPage';
import About from './About'
import Features from './Features';
import Register from './Register';
import Profile from './Profile';
import Contacts from './Contacts';
import Messaging from './Messaging';
export default class App extends React.Component {

	constructor(props) {
		super(props);
		this.state = {
		}
		this.handleUserChange = this.handleUserChange.bind(this);
	}

	handleUserChange(username) {
		this.setState({username: username});
	}

	render() {
		return (
			<div data-testid="main" id="main" className="App">
				<Router>
					<Switch>
						<Route
							exact
							path="/"
							render={() => (
								<LandingPage />
							)}
						/>
						<Route
							exact
							path="/about"
							render={() => (
								<About />
							)}
						/>
						<Route
							exact
							path="/features"
							render={() => (
								<Features />
							)}
						/>
						<Route
							exact
							path="/register"
							render={() => (
								<Register />
							)}
						/>
						<Route
							exact
							path="/profile"
							render={() => (
								<Profile username = {this.state.username} />
							)}
						/>
						<Route
							exact
							path="/contacts"
							render={() => (
								<Contacts username = {this.state.username}/>
							)}
						/>
						<Route
							exact
							path="/messaging"
							render={() => (
								<Messaging />
							)}
						/>
					</Switch>
				</Router>
			</div>
		);
	}

}

