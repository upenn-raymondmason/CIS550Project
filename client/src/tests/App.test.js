import {
  act, fireEvent, render, screen, waitFor, waitForElement,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from '../pages/App';
import About from '../pages/About';
import Features from '../pages/Features';
import LandingPage from '../pages/LandingPage';
import Profile from '../pages/Profile';
import Register from '../pages/Register';
import fetchMock from 'jest-fetch-mock';
require('jest-fetch-mock').enableMocks();
const renderer = require('react-test-renderer');

beforeEach(() => {
  fetch.resetMocks();
});

// snapshot tests
describe('Snapshot tests for each of the pages', () => {
  test('Landing Page matches snapshot', () => {
    const component = renderer.create(<LandingPage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('About matches snapshot', () => {
    const component = renderer.create(<About />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Features matches snapshot', () => {
    const component = renderer.create(<Features />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Profile matches snapshot', () => {
    const component = renderer.create(<Profile />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Register matches snapshot', () => {
    const component = renderer.create(<Register />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// Test Components Presence
describe('the initial page loaded correctly', () => {
  test('Correct App page', () => {
    render(<App />);
    const divElement = screen.getByTestId('main');
    expect(divElement).toBeInTheDocument();
  });
  test('Correct landing page', () => {
    render(<LandingPage />);
    const divElement = screen.getByTestId('landing');
    expect(divElement).toBeInTheDocument();
  });
});

// Test routing
/* describe('Routing is working correctly', () => {
  test('Page routes to About', async () => {
    render(<App />);
    fireEvent.click(screen.getByText(/About/));
    let divEle = null;
    await waitFor(() => {divEle = screen.getByTestId('AboutBox');});
    expect(divEle).toBeInTheDocument();
    });
}); */
