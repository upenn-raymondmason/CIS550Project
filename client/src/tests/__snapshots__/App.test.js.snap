// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for each of the pages About matches snapshot 1`] = `
<div
  className="App"
  data-testid="AboutBox"
  id="AboutBox"
>
  <nav
    className="NavBarItems"
  >
    <div
      className="size"
      style={
        Object {
          "transform": "scale(0.75)",
        }
      }
    >
      <img
        alt="logo"
        className="Navbar-icon"
        src="penn_book_menu_icon.svg"
      />
    </div>
    <h1
      className="NavBar-logo"
    >
      PennBook
    </h1>
    <div
      className="Navbar-icon"
    />
    <ul
      className="NavBar-menu"
    >
      <li>
        <a
          className="nav-links"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/features"
        >
          Features
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/register"
        >
          Register
        </a>
      </li>
    </ul>
  </nav>
  <br />
  <h1>
    About
  </h1>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
</div>
`;

exports[`Snapshot tests for each of the pages Features matches snapshot 1`] = `
<div
  className="App"
>
  <nav
    className="NavBarItems"
  >
    <div
      className="size"
      style={
        Object {
          "transform": "scale(0.75)",
        }
      }
    >
      <img
        alt="logo"
        className="Navbar-icon"
        src="penn_book_menu_icon.svg"
      />
    </div>
    <h1
      className="NavBar-logo"
    >
      PennBook
    </h1>
    <div
      className="Navbar-icon"
    />
    <ul
      className="NavBar-menu"
    >
      <li>
        <a
          className="nav-links"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/features"
        >
          Features
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/register"
        >
          Register
        </a>
      </li>
    </ul>
  </nav>
  <br />
  <h1>
    Features
  </h1>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
</div>
`;

exports[`Snapshot tests for each of the pages Landing Page matches snapshot 1`] = `
<div
  className="App"
  data-testid="landing"
>
  <nav
    className="NavBarItems"
  >
    <div
      className="size"
      style={
        Object {
          "transform": "scale(0.75)",
        }
      }
    >
      <img
        alt="logo"
        className="Navbar-icon"
        src="penn_book_menu_icon.svg"
      />
    </div>
    <h1
      className="NavBar-logo"
    >
      PennBook
    </h1>
    <div
      className="Navbar-icon"
    />
    <ul
      className="NavBar-menu"
    >
      <li>
        <a
          className="nav-links"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/features"
        >
          Features
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/register"
        >
          Register
        </a>
      </li>
    </ul>
  </nav>
  <div
    style={
      Object {
        "alignSelf": "right",
        "float": "right",
        "marginRight": "5vw",
      }
    }
  >
    <div
      className="LoginForm"
    >
      <h2
        className="LoginForm-title"
      >
        Login
      </h2>
      <br />
      <form
        onSubmit={[Function]}
      >
        <input
          className="LoginForm-input"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
        <br />
        <input
          className="LoginForm-input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <br />
        <input
          className="LoginForm-button"
          type="submit"
          value="Login"
        />
      </form>
    </div>
  </div>
  <br />
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
</div>
`;

exports[`Snapshot tests for each of the pages Profile matches snapshot 1`] = `
<div>
  <nav
    className="NavBarItems"
  >
    <div
      className="size"
      style={
        Object {
          "transform": "scale(0.75)",
        }
      }
    >
      <img
        alt="logo"
        className="Navbar-icon"
        src="penn_book_menu_icon.svg"
      />
    </div>
    <h1
      className="NavBar-logo"
    >
      PennBook
    </h1>
    <div
      className="Navbar-icon"
    />
    <ul
      className="NavBar-menu"
    >
      <li>
        <a
          className="nav-links"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/features"
        >
          Features
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/register"
        >
          Register
        </a>
      </li>
    </ul>
  </nav>
  <h1>
    Profile
  </h1>
</div>
`;

exports[`Snapshot tests for each of the pages Register matches snapshot 1`] = `
<div
  className="App"
>
  <nav
    className="NavBarItems"
  >
    <div
      className="size"
      style={
        Object {
          "transform": "scale(0.75)",
        }
      }
    >
      <img
        alt="logo"
        className="Navbar-icon"
        src="penn_book_menu_icon.svg"
      />
    </div>
    <h1
      className="NavBar-logo"
    >
      PennBook
    </h1>
    <div
      className="Navbar-icon"
    />
    <ul
      className="NavBar-menu"
    >
      <li>
        <a
          className="nav-links"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/features"
        >
          Features
        </a>
      </li>
      <li>
        <a
          className="nav-links"
          href="/register"
        >
          Register
        </a>
      </li>
    </ul>
  </nav>
  <div
    style={
      Object {
        "alignSelf": "right",
        "float": "right",
        "marginRight": "5vw",
      }
    }
  >
    <div
      className="SignupForm"
    >
      <h2
        className="SignupForm-title"
      >
        Sign Up
      </h2>
      <br />
      <form
        onSubmit={[Function]}
      >
        <input
          className="SignupForm-input"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
        <br />
        <input
          className="SignupForm-input"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
        <br />
        <input
          className="SignupForm-input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <br />
        <input
          className="SignupForm-button"
          id="SignupForm-button-id"
          type="submit"
          value="Sign up"
        />
      </form>
    </div>
  </div>
  <br />
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
</div>
`;
